src/OpenRiaServices.Hosting.AspNetCore/Framework/AspNetCore/OpenRiaServicesServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,107 @@
+{
+    /// <summary>
+    /// Extension methods for registering OpenRiaServices services in an <see cref="IServiceCollection"/>
+    /// </summary>
+    public static class OpenRiaServicesServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Adds services required for using OpenRiaServices
+        /// </summary>
+        public static OpenRiaServicesOptionsBuilder AddOpenRiaServices(this IServiceCollection services)
+        {
+            ArgumentNullException.ThrowIfNull(services);
+
+            services.AddSingleton<OpenRiaServicesEndpointDataSource>();
+            services.AddSingleton(services);
+
+            return new OpenRiaServicesOptionsBuilder(services);
+        }
+
+        /// <summary>
+        /// Adds services required for using OpenRiaServices and configures the options
+        /// </summary>
+        public static OpenRiaServicesOptionsBuilder AddOpenRiaServices(this IServiceCollection services, Action<OpenRiaServicesOptions> configure)
+        {
+            ArgumentNullException.ThrowIfNull(services);
+
+            services.AddOpenRiaServices();
+            services.Configure(configure);
+
+            return new OpenRiaServicesOptionsBuilder(services);
+        }
+
+        /// <summary>
+        /// Registers <typeparamref name="TService"/> as a transient service in <paramref name="services"/>
+        /// </summary>
+        public static IServiceCollection AddDomainService<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods | DynamicallyAccessedMemberTypes.PublicConstructors)] TService>(this IServiceCollection services) where TService : DomainService
+        {
+            return AddDomainService<TService>(services, ServiceLifetime.Transient);
+        }
+
+        /// <summary>
+        /// Registers <typeparamref name="TService"/> as a service with <paramref name="serviceLifetime"/> in <paramref name="services"/>
+        /// </summary>
+        public static IServiceCollection AddDomainService<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods | DynamicallyAccessedMemberTypes.PublicConstructors)] TService>(this IServiceCollection services, ServiceLifetime serviceLifetime) where TService : DomainService
+        {
+            services.Add(new ServiceDescriptor(typeof(TService), typeof(TService), serviceLifetime));
+            return services;
+        }
+
+        // TODO:
+        // - From or In Assemblies
+        // [RequiresUnreferencedCode("DomainService types are loaded dynamically and may be trimmed.")]
+        /// <summary>
+        /// Registers public <see cref="DomainService"/>s marked with <see cref="EnableClientAccessAttribute"/> in <paramref name="assemblies"/> for dependency injection
+        /// </summary>
+        public static IServiceCollection AddDomainServicesFromAssemblies(this IServiceCollection services, IEnumerable<Assembly> assemblies, ServiceLifetime serviceLifetime = ServiceLifetime.Transient)
```

Comment:
// TODO: handle naming
        // - From or In Assemblies


Should it be `AddDomainServicesInAssemblies` instead if so then change `AddDomainServicesFromAssembly` => `AddDomainServicesInAssembly` ?
```suggestion
        // [RequiresUnreferencedCode("DomainService types are loaded dynamically and may be trimmed.")]
        /// <summary>
        /// Registers public <see cref="DomainService"/>s marked with <see cref="EnableClientAccessAttribute"/> in <paramref name="assemblies"/> for dependency injection
        /// </summary>
        public static IServiceCollection AddDomainServicesFromAssemblies(this IServiceCollection services, IEnumerable<Assembly> assemblies, ServiceLifetime serviceLifetime = ServiceLifetime.Transient)
```

Reply:
Change to AddDomainServices